### DISCUSSION TITLE
MySQL - 'Failed to find inserted item' when inserting an item which has an OffsetDateTime primary key

### DISCUSSION THREAD

**Discussion by muttleyxd** [SeaQL/sea-orm#1764] [General]
## Description

As the title states - I get an error saying 'Failed to find inserted item' when inserting an item which  has an OffsetDateTime primary key.
This happens only on MySQL. It works fine on SQLite, I didn't test with Postgres.

## Steps to Reproduce

Basically you need an entity, which has OffsetDateTime (from time library) as primary key. Insertion will fail with mentioned error.

Refer to my repository (link below).
```
// unit_state.rs:
use sea_orm::entity::prelude::*;
use time::OffsetDateTime;

#[derive(Clone, Debug, PartialEq, Eq, DeriveEntityModel)]
#[sea_orm(table_name = "unit_state")]
pub struct Model {
    #[sea_orm(primary_key, auto_increment = false)]
    pub created_date: OffsetDateTime,
}

impl ActiveModelBehavior for ActiveModel {}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

// main.rs:
#[derive(Iden)]
pub enum UnitState {
    Table,
    CreatedDate,
}

async fn create_tables(db: &DatabaseConnection) -> Result<(), DbErr> {
    let manager = SchemaManager::new(db);

    manager
        .create_table(
            Table::create()
                .table(UnitState::Table)
                .if_not_exists()
                .col(
                    ColumnDef::new(UnitState::CreatedDate)
                        .date_time()
                        .primary_key()
                        .not_null(),
                )
                .to_owned(),
        )
        .await?;

    Ok(())
}

async fn drop_tables(db: &DatabaseConnection) -> Result<(), DbErr> {
    let manager = SchemaManager::new(db);

    manager
        .drop_table(Table::drop().table(UnitState::Table).to_owned())
        .await?;

    Ok(())
}

#[tokio::main]
async fn main() {
    let database_url = "mysql://test:test@localhost:23456/test";
    let db = sea_orm::Database::connect(database_url)
        .await
        .expect("Database Connected");

    create_tables(&db).await.unwrap();

    entities::unit_state::ActiveModel {
        created_date: Set(OffsetDateTime::now_utc()),
    }
    .insert(&db)
    .await
    .unwrap(); // DbErr here!

    drop_tables(&db).await.unwrap();
}
```

Basically you need an entity, which has primary key constructed with OffsetDateTime.

### Expected Behavior

Item gets inserted into the database. Return value is an ActiveModel object.

### Actual Behavior

Item gets inserted into the database, but insert function returns `Err(RecordNotFound("Failed to find inserted item"))`

### Reproduces How Often

Looks like 100% to me.

### Workarounds

This happens only when using ActiveModel - workaround proposed in #244 (use `Entity::insert()`) works fine.

## Reproducible Example

I prepared a small repository to help reproducing this - it contains scripts to create the database and a small piece of Rust code to reproduce the issue with sea-orm v0.12.0-rc3. It also reproduces with v0.11.

https://github.com/muttleyxd/sea-orm-insertion-failure

## Versions

Reproduces both with newest 0.11 and 0.12.0-rc.3.
Reproduced on Debian and Arch Linux.

```
$ rustc --version
rustc 1.70.0 (90c541806 2023-05-31)

$  cargo tree | grep sea-
sea-orm-insertion-failure v0.1.0 (/home/muttley/git/sea-orm-insertion-failure)
├── sea-orm v0.12.0-rc.4
│   ├── sea-orm-macros v0.12.0-rc.4 (proc-macro)
│   ├── sea-query v0.29.1
│   │   ├── sea-query-derive v0.4.0 (proc-macro)
│   ├── sea-query-binder v0.4.0
│   │   ├── sea-query v0.29.1 (*)
├── sea-orm-migration v0.12.0-rc.4
│   ├── sea-orm v0.12.0-rc.4 (*)
│   ├── sea-orm-cli v0.12.0-rc.4
│   │   ├── sea-schema v0.12.0-rc.2
│   │   │   ├── sea-query v0.29.1 (*)
│   │   │   └── sea-schema-derive v0.2.0-rc.2 (proc-macro)
│   ├── sea-schema v0.12.0-r
```

**Comment by tyt2y3** [comment]
If you are only using MySQL and SQLite, I would suggest using `varchar` as the underlying column type.
This would unify their behaviour.
